### ISSUE TITLE
Length of returned list was different than expected - expected 1 items, got 2

### ISSUE THREAD

**Issue by varayal** [taverntesting/tavern#303]
Can anyone indicate me how can I match the first element of the array Categories -> "code: 1"

Actually, I got this error:

```

 failed:
E           - Value mismatch in body: Length of returned list was different than expected - expected 1 items, got 2 (expected["data"]["Categories"] = '[{'code': <tavern.util.loader.AnythingSentinel object at 0x1068f7a90>}]
```

The JSON is 

```
    "data": {
        "title": "hello",
        "Categories": [
            {
                "code": "1",
                "name": "iphone"
            },
            {
                "code": "2",
                "name": "iphone X"
            }
```

the tavern test is

```
    response:
      body:
        data:
          title: "hello"
          Categories:
           -
              code: "1"
```

**Comment by michaelboulton** [comment]
At the moment this is not really possible ( see https://github.com/taverntesting/tavern/issues/133 ), I am currently working on a fix for it in #289

**Comment by michaelboulton** [comment]
Just going to close this to keep discussion in one issue https://github.com/taverntesting/tavern/issues/133#issuecomment-481774309

For the time being you can try using this branch https://github.com/taverntesting/tavern/pull/162 which I am keeping up to date with the master branch
